benchmarks/run_release_benchmark.py

```diff
@@ -0,0 +1,212 @@
+
+    client.connect(hostname=public_ip_address, username="ubuntu", pkey=key)
+
+    print("Connected to remote EC2 instance...")
+    print("Cloning serve repo...")
+    command = 'git clone https://github.com/pytorch/serve.git && cd serve && git checkout {}'.format(branch)
+
+    print("Executing command  : {}".format(command))
+    stdin, stdout, stderr = client.exec_command(command)
+
+    exit_status = stdout.channel.recv_exit_status()
+
+    if exit_status != 0:
+        print(stdout.read())
+        print(stderr.read())
+
+    print("TorchServe repo cloned.")
+    print("Creating docker image...")
+    command = 'cd serve/docker && ./build_image.sh' \
+              ' {docker_type} --branch_name {branch_name} --tag pytorch/torchserve:{tag_name}' \
+        .format(docker_type="--gpu" if instance_type == "gpu" else "",
+                branch_name=branch,
+                tag_name=instance_type
+                )
+    print("Executing command  : {}".format(command))
+    stdin, stdout, stderr = client.exec_command(command)
+    exit_status = stdout.channel.recv_exit_status()
+
+    if exit_status != 0:
+        print(stdout.read())
+        print(stderr.read())
+
+    print("Docker image creation completed.")
+    print("Installing benchmark dependencies...")
+    command = 'cd serve/benchmarks' \
+              ' && pip install -U -r requirements-ab.txt' \
+              ' && sudo apt-get install -y apache2-utils'
+    print("Executing command  : {}".format(command))
+    stdin, stdout, stderr = client.exec_command(command)
+    exit_status = stdout.channel.recv_exit_status()
+    if exit_status != 0:
+        print(stdout.read())
+        print(stderr.read())
+
+    benchmark_config_path = "preset_configs/{model_name}/{instance_type}/{model_mode}_batch_{batch_size}.json".format(
+        model_name=model_name,
+        instance_type=instance_type,
+        model_mode=model_mode,
+        batch_size=batch_size
+    )
+
+    print("Benchmark dependency installation completed.")
+    print("Running benchmark...")
+    command = 'cd serve/benchmarks && python benchmark-ab.py --config {}'.format(benchmark_config_path)
+    print("Executing command  : {}".format(command))
+    stdin, stdout, stderr = client.exec_command(command)
+    exit_status = stdout.channel.recv_exit_status()
+
+    if exit_status != 0:
+        print(stdout.read())
+        print(stderr.read())
+
+    print("Benchmark execution completed... Collecting results...")
+    command = 'cat /tmp/benchmark/ab_report.csv'
+    print("Executing command  : {}".format(command))
+    stdin, stdout, stderr = client.exec_command(command)
+
+    report_path = '/tmp/benchmark'
+    if not os.path.exists(report_path):
+        os.makedirs(report_path)
+
+    report_file = '{}/{}_{}_{}.csv'.format(report_path, model_name, model_mode, batch_size)
+
+    with open(report_file, 'w') as rf:
+        for line in stdout.readlines():
+            rf.write(line)
+
+    client.close()
```

Comment:
The preferred way to run this would be to have use user-data based startup script instead of do this through SSH.  This was you dont even need a key file. 

More details at - https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/user-data.html

Will be happy to go over the details offline.

Reply:
Done. Have added details about other changes we discussed offline in the PR's main comment.